src/main/java/net/fabricmc/loader/EntrypointStorage.java

```diff
@@ -145,20 +157,39 @@ boolean hasEntrypoints(String key) {
 					results.add(result);
 				}
 			} catch (Throwable t) {
-				errors.add(t);
+				errors.put(entry, t);
 			}
 		}
 
 		if (!errors.isEmpty()) {
 			EntrypointException e = new EntrypointException("Could not look up entries for entrypoint " + key + "!");
 
-			for (Throwable suppressed : errors) {
-				e.addSuppressed(suppressed);
+			for (Map.Entry<Entry, Throwable> suppressed : errors.entrySet()) {
+				e.addSuppressed(new EntrypointException("Caused by mod: '" + suppressed.getKey().getModContainer().getMetadata().getId() + "' ", suppressed.getValue()));
 			}
 
 			throw e;
 		} else {
 			return results;
 		}
 	}
+
+	ModContainer getContainerFromEntrypoint(Object entrypoint) {
+		for (Map.Entry<String, List<Entry>> entrypointList : entryMap.entrySet()) {
```

Comment:
Will this be too slow time-wise? How costly memory-wise is it to track in a hash map instead?